### PR TITLE
BUG: Fix structured_to_unstructured on single-field types

### PR DESCRIPTION
**Pull request by eric-wieser**
Previously a single-field type would decay, which is undesirable.

The included test previously did not pass

### CODE REVIEW

File: numpy/lib/recfunctions.py [hunk 1]
```diff
@@ -980,7 +980,7 @@ def structured_to_unstructured(arr, dtype=None, copy=False, casting='unsafe'):
     arr = arr.astype(packed_fields, copy=copy, casting=casting)
 
     # finally is it safe to view the packed fields as the unstructured type
-    return arr.view((out_dtype, sum(counts)))
+    return arr.view((out_dtype, (sum(counts),)))
```
Comment (eric-wieser) [inline numpy/lib/recfunctions.py]: This is a strong argument for why #13326 is a good idea

File: numpy/lib/tests/test_recfunctions.py [hunk 2]
```diff
@@ -214,6 +214,8 @@ def test_structured_to_unstructured(self):
                      dtype=[('x', 'i4'), ('y', 'f4'), ('z', 'f8')])
         out = np.mean(structured_to_unstructured(b[['x', 'z']]), axis=-1)
         assert_equal(out, np.array([ 3. ,  5.5,  9. , 11. ]))
+        out = np.mean(structured_to_unstructured(b[['x']]), axis=-1)
+        assert_equal(out, np.array([ 1. ,  4. ,  7. , 10. ]))
```
Comment (eric-wieser) [inline numpy/lib/tests/test_recfunctions.py]: Previously returned the scalar `5.5`

**Comment by eric-wieser** [comment]
Rebased for ease of backporting.

**Comment by charris** [comment]
@ahaldane Thoughts?

File: numpy/lib/tests/test_recfunctions.py (hunk 2 above)
Comment (eric-wieser) [reply numpy/lib/tests/test_recfunctions.py]: If someone actually wants the result `5.5`, they should just have been using `np.mean(b['x'], axis=-1)`

**Comment by ahaldane** [comment]
Just got to this. Yes, it seems like the right change to make.

I think it is worth a release note since it is technically a backward incompatible change. I just hope no one has depended on this yet! (seems unlikely since it a new function).

After that, LGTM, I'd be happy to merge.

**Comment by ahaldane** [comment]
Oh I see, you are planning to put it in 1.16.3. In that case we can just go ahead and merge right?

**Comment by eric-wieser** [comment]
If its not too late to put in 1.16.3...

**Comment by ahaldane** [comment]
Oh sorry, I was away the last week so may have caused the cutoff date to be missed. It would be nice to get it in 1.16.3...

**Comment by eric-wieser** [comment]
@charris, this should get 1.16.4 notes I assume?

**Comment by eric-wieser** [comment]
Note that our release notes for 1.16.0 barely mention `structured_to_unstructured` in any detail.

>  think it is worth a release note since it is technically a backward incompatible change

Every bug fix is a backwards-incompatible change for those depending on it. I'd perhaps argue that the changelog is sufficient here, but will let @charris arbitrate.

**Comment by charris** [comment]
> this should get 1.17.0 / 1.16.4 notes I assume?

Just 1.17.0, then do 1.16.4 for the backport. Note that there is no 1.17.0 in 1.16.x, so the file will need removal.

**Comment by eric-wieser** [comment]
Updated with a 1.17 release note, 1.16 release note created in #13396

**Comment by mattip** [comment]
Thanks Eric

**Pull request merged**